src/native/minipal/file.c

```diff
@@ -0,0 +1,198 @@
+    
+    // Now the path will be normalized
+
+    size_t prefixLength = sizeof(ExtendedPrefix) / sizeof(WCHAR) - 1;
+    size_t bufferLength = length + prefixLength + 1;
+    WCHAR* buffer = (WCHAR*)malloc(bufferLength * sizeof(WCHAR));
+
+    if (!buffer)
+        return NULL;
+
+    DWORD retSize = GetFullPathNameW(path, (DWORD)(bufferLength - prefixLength), buffer + prefixLength, NULL);
+
+    if (retSize > bufferLength - prefixLength)
+    {
+        free(buffer);
+        bufferLength = retSize + prefixLength;
+        buffer = (WCHAR*)malloc(bufferLength * sizeof(WCHAR));
+
+        if (!buffer)
+            return NULL;
+
+        retSize = GetFullPathNameW(path, (DWORD)(bufferLength - prefixLength), buffer + prefixLength, NULL);
+    }
+
+    if (retSize <= 0)
+    {
+        free(buffer);
+        return NULL;
+    }
+
+    memcpy(buffer, ExtendedPrefix, bufferLength * sizeof(WCHAR));
+
+    return buffer;
+}
+#else // TARGET_WINDOWS
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdlib.h>
+#include <minipal/utf8.h>
+#include <minipal/strings.h>
+
+#if HAVE_STAT64
+#define stat_ stat64
+#define fstat_ fstat64
+#define lstat_ lstat64
+#else /* HAVE_STAT64 */
+#define stat_ stat
+#define fstat_ fstat
+#define lstat_ lstat
+#endif  /* HAVE_STAT64 */
+
+/* Magic number explanation:
+
+   To 1970:
+   Both epochs are Gregorian. 1970 - 1601 = 369. Assuming a leap
+   year every four years, 369 / 4 = 92. However, 1700, 1800, and 1900
+   were NOT leap years, so 89 leap years, 280 non-leap years.
+   89 * 366 + 280 * 365 = 134774 days between epochs. Of course
+   60 * 60 * 24 = 86400 seconds per day, so 134774 * 86400 =
+   11644473600 = SECS_BETWEEN_1601_AND_1970_EPOCHS.
+
+   To 2001:
+   Again, both epochs are Gregorian. 2001 - 1601 = 400. Assuming a leap
+   year every four years, 400 / 4 = 100. However, 1700, 1800, and 1900
+   were NOT leap years (2000 was because it was divisible by 400), so
+   97 leap years, 303 non-leap years.
+   97 * 366 + 303 * 365 = 146097 days between epochs. 146097 * 86400 =
+   12622780800 = SECS_BETWEEN_1601_AND_2001_EPOCHS.
+
+   This result is also confirmed in the MSDN documentation on how
+   to convert a time_t value to a win32 FILETIME.
+*/
+static const int64_t SECS_BETWEEN_1601_AND_1970_EPOCHS = 11644473600LL;
+static const int64_t SECS_TO_100NS = 10000000; /* 10^7 */
+
+static uint64_t UnixTimeToWin32FileTime(struct timespec ts)
+{
+    return (uint64_t)(((int64_t)ts.tv_sec + SECS_BETWEEN_1601_AND_1970_EPOCHS) * SECS_TO_100NS +
+        (ts.tv_nsec / 100));
+}
+
+#endif // TARGET_WINDOWS
+
+bool minipal_file_get_attributes_utf16(const char16_t* path, minipal_file_attr_t* attributes)
+{
+#ifdef TARGET_WINDOWS
+    if (!path || !attributes)
+        return false;
+
+    WCHAR* extendedPath = NormalizePath(path);
+    if (extendedPath)
+        path = extendedPath;
+
+    WIN32_FILE_ATTRIBUTE_DATA faData;
+    bool ret = GetFileAttributesExW(path, GetFileExInfoStandard, &faData);
```

Comment:
```suggestion
    BOOL ret = ::GetFileAttributesExW(path, GetFileExInfoStandard, &faData);
```

Reply:
`::` is C++ syntax and not available in C.

I'm not sure whether `bool` from `stdbool.h` can convert fine from `BOOL`. Maybe `!!` is required.

The whole implementation may be removed per https://github.com/dotnet/runtime/pull/116096#discussion_r2114134664, so it may not be worthy to review it now.